src/Core/Addon/Theme/ThemeRepository.php

```diff
@@ -51,22 +51,27 @@ public function __construct(ConfigurationInterface $configuration, Filesystem $f
 
     public function getInstanceByName($name)
     {
-        $dir = $this->appConfiguration->get('_PS_ALL_THEMES_DIR_').$name;
-        $themeCachePath = $this->appConfiguration->get(
-                '_PS_CACHE_DIR_'
-            ) . 'themes/' . $name;
+        $dir = $this->appConfiguration->get('_PS_ALL_THEMES_DIR_') . $name;
 
+        $confDir = $this->appConfiguration->get('_PS_CONFIG_DIR_') . 'themes/' . $name;
+        $jsonConf = '';
         if ($this->shop) {
-            $jsonConf = $themeCachePath . '/shop' . $this->shop->id . '.json';
-        } else {
-            $jsonConf = $themeCachePath . '/theme.json';
+            $jsonConf = $confDir . '/shop' . $this->shop->id . '.json';
         }
 
         if ($this->filesystem->exists($jsonConf)) {
             $data = $this->getConfigFromFile($jsonConf);
         } else {
-            $data = $this->getConfigFromFile($dir.'/config/theme.yml');
-            $this->filesystem->dumpFile($jsonConf, json_encode($data));
```

Comment:
$this->filesystem->dumpFile($jsonConf, json_encode($data), 0777); ? (since it's atomic, it's better than the classical file_put_contents)

Reply:
Here again, I just reused the old code, and tried to avoid altering the behavior. I'm open to suggestions though. So if you thing this method is better, I'll change it (please confirm).

Reply:
yes dumpFile seems to be better here, as it's atomic and basically already takes care of the directory creation